test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -6693,6 +6713,27 @@ def manager(self, cmd, obj_type, attrib=None, id=None, extend=None,
         :type logerr: bool
         :raises: PbsManagerError
         """
+        skipmsg = 'ncpus requested are not same as available'
+        if cmd == MGR_CMD_SET and id is not None:
+            if obj_type == NODE:
+                for cmom, momobj in self.moms.items():
+                    cpuset_nodes = []
+                    if momobj.is_cpuset_mom():
+                        vnodes = self.status(NODE)
+                        del vnodes[0]  # remove natural node
+                        for vn in vnodes:
+                            at = 'resources_available.ncpus'
+                            if 'resources_available.ncpus' in attrib:
+                                if int(attrib[at]) != int(vn[at]):
+                                    self.skipTest(skipmsg)
+                                else:
+                                    del attrib['resources_available.ncpus']
+                        if 'resources_available.mem' in attrib:
```

Comment:
I would suggest put this check before we check ncpus, it will save one call to pbsnodes if mem is requested